shared/src/metabase/mbql/util.cljc

```diff
@@ -459,14 +459,19 @@
        (not (time-field? field))))
 
 (defn datetime-arithmetics?
-  "Is a given artihmetics clause operating on datetimes?"
+  "Does a given artihmetics clause return a Datetime type?"
   [clause]
   (mbql.match/match-one clause
-    #{:interval :relative-datetime}
+    :relative-datetime
```

Comment:
We probably should have `datetimeAdd and datetimeSubtract` here too right?

Reply:
Yes. Thinking about this, I just discovered an issue with the use of `datetime-arithmetics?` in `infer-expression-type` ([here](https://github.com/metabase/metabase/blob/5ded6a0639df5a69bffbe87e0627edeac6d0a8e2/src/metabase/query_processor/middleware/annotate.clj#L162)).

`datetime-arithmetics?` implies the return type is always a `:type/Datetime`, but `datetimeAdd` etc can return a value of type `:type/Date` or `:type/Datetime`. Is this causing any bugs at the moment?

Reply:
![Screen Shot 2022-11-08 at 19 06 58](https://user-images.githubusercontent.com/25661381/200559821-d9ae687a-7c3c-4501-8f03-4369fda18746.png)
yes, UI will format the date column as a datetime.

Maybe we have to look into the expression type for this.

Reply:
Sorry, I totally mixed things up here. `datetimeAdd` always returns a `:type/DateTime`, and never `:type/Date`. My mistake. That surprised me, but I'm guessing that is intentional?

Reply:
I've added datetimeAdd and datetimeSubtract here as you originally suggested in your first comment.